server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1042,7 +1044,7 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
         } else {
             try {
                 numericValue = fieldType().type.parse(parser, coerce.value());
-            } catch (IllegalArgumentException e) {
+            } catch (IllegalArgumentException | JsonParseException e) {
```

Comment:
I haven't tested this, but I wanted to check that this will not introduce too much leniency? In particular, I'm wondering what happens if we try to pass a JSON object to a numeric field with `ignore_malformed` enabled.

Reply:
The way the parser works, anything that's not a number or string token after the field name should throw an error. If "swallowing" that token with ignore_malformed leads to unparseable xcontent afterwards we will still throw an error. I think that's what should happen.

Reply:
Thanks for clarifying. What would you think about adding a check whether the token is a value, as in if (ignoreMalformed.value() && parser.currentToken().isValue()) { ... }? That way a user would get a clear, localized error like "Current token (START_OBJECT) not numeric", as opposed to a more difficult one to track down like "Malformed content, found extra data after parsing: END_OBJECT".

Reply:
The problem with that is that the code actually throwing the exception is in the jackson library. The only way I could think of to do this is either have some logic taking apart the exception or pre-checking the token in NumberFieldMapper. I'd try to avoid the former, was the later what you had in mind? We'd need to test for (and accept) all value tokens here basically?